core/src/test/java/org/testcontainers/utility/DockerImageNameTest.java

```diff
@@ -0,0 +1,18 @@
+package org.testcontainers.utility;
+
+import org.junit.Test;
+
+public class DockerImageNameTest {
+
+    @Test
+    public void validNames() {
+        DockerImageName.validate("myname:latest"); // no repo
+        DockerImageName.validate("repo/my-name:1.0"); // no repo
+        DockerImageName.validate("repo.foo.com:1234/my-name:1.0"); // no repo
+    }
```

Comment:
That's interesting - probably worth thinking about moving towards NameParser in the near future if it will encapsulate the validation we want in a single place.

However, I think I want to continue to insist that the tag is always present for explicit versioning. Defaulting to 'latest' seems too dangerous in a testing context, so I'd like users to have to consciously set it if we allow it at all ;)

Given that this PR is functional I think I'll accept as-is, with the awareness that we could investigate using NameParser some time instead!

Thanks
Rich